Restructure Node SDK docs. (#5710)
getsentry/sentry-docs: src/platform-includes/configuration/integrations/default-integrations/node.mdx

System integrations are enabled by default to integrate into the standard library or the interpreter itself. They are documented so you can be both aware of what they do and disable them if they cause issues.

## Enabled by Default

### InboundFilters

_Import name: `Sentry.Integrations.InboundFilters`_

> This integration allows developers to ignore specific errors based on the error message or the URL from which the exception originated.

Reviewer: ```suggestion
This integration allows you to ignore specific errors based on the error message or the URL from which the exception originated.
```

To configure it, use the `ignoreErrors`, `denyUrls`, and `allowUrls` SDK options directly.

> (Keep in mind that `denyUrls` and `allowUrls` only work for captured exceptions, not raw message events.)

Reviewer: ```suggestion
Keep in mind that `denyUrls` and `allowUrls` only work for captured exceptions, not raw message events.
```

### FunctionToString

_Import name: `Sentry.Integrations.FunctionToString`_

This integration allows the SDK to provide original function and method names, even when those functions or methods
are wrapped by our error or breadcrumb handlers.

## Node-specific

### Console

_Import name: `Sentry.Integrations.Console`_

> This integration wraps the `console` module in order to record all of its calls as breadcrumbs.

Reviewer: ```suggestion
This integration wraps the `console` module to record all of its calls as breadcrumbs.
```

### Http

_Import name: `Sentry.Integrations.Http`_

> This integration wraps `http` and `https` modules to capture all network requests as breadcrumbs and/or tracing spans.

Reviewer: ```suggestion
This integration wraps the `http` and `https` modules to capture all network requests as breadcrumbs and/or tracing spans.
```

Available options:

```javascript
{
  breadcrumbs: boolean; // default: true
  tracing: boolean; // default: false
}
```

### OnUncaughtException

_Import name: `Sentry.Integrations.OnUncaughtException`_

This integration attaches a global uncaught exception handler. It can be modified to provide a custom shutdown function.
The `onFatalError` option is meant to perform a cleanup before the process exits, not fully prevent it from exiting.

> <Alert level="warning" title="Note">

Reviewer: Does this need to be a warning? Are you breaking something? If not, let's bump this down to a regular note.
```suggestion
<Alert level="warning">
```

Be aware that if you overwrite this setting, you will lose the default implementation, which handles draining queued events before exiting. If you want to examine how it works in order to be able to recreate it in your code, see the implementation of the [`logAndExitProcess`](https://github.com/getsentry/sentry-javascript/blob/master/packages/node/src/integrations/utils/errorhandling.ts) function.

</Alert>

Available options:

```javascript
{
  onFatalError: (firstError: Error, secondError?: Error) => void;
}
```

### OnUnhandledRejection

_Import name: `Sentry.Integrations.OnUnhandledRejection`_

> This integration attaches a global unhandled rejection handler. By default, all unhandled rejections trigger a warning and log the error. You can change this behavior using the `mode` option, which is in line with Node's CLI options: https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode.

Reviewer: Attempting to simplify the language a bit

```suggestion
This integration attaches a global unhandled rejection handler. By default, all unhandled rejections trigger a warning and log the error. You can change this behavior using the `mode` option, which is works with Node's CLI options: https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode.
```

Available options:

```javascript
{
  mode: "none" | "warn" | "strict"; // default: "warn"
}
```

### ContextLines

_Import name: `Sentry.Integrations.ContextLines`_

_(Available in version 6.18.0 and above)_

> This integration adds source file context to stack frames for captured exceptions

Reviewer: ```suggestion
This integration adds source file context to stack frames for captured exceptions.
```

Available options:

```javascript
{
  // The number of context lines to include with each frame
  frameContextLines: number;
}
```

### LinkedErrors

_Import name: `Sentry.Integrations.LinkedErrors`_

> This integration allows you to configure linked errors. They'll be recursively read up to a specified limit and lookup will be performed by a specific key. By default, the limit is set to 5 and the key used is `"cause"`.

Reviewer: _5_ either needs to be written out or put in code font to denote it as a setting. Also, in the JS version of this there are no quotation marks on `cause`. Just highlighting that in case it's not intentional. Lastly, is this node-specific if it's also available as a default integration for JS?

```suggestion
This integration allows you to configure linked errors. They'll be recursively read up to a specified limit, and lookup will be performed by a specific key. By default, the limit is set to 5 and the key used is `"cause"`.
```

Author: - Updated `5` to use code font.
- `"cause"` should be the correct one, as it's a `string`. Updated the JS docs.
- Yes, `LinkedErrors` is not Node-specific, moved it out. 👍

Available options:

```javascript
{
  key: string; // default: "cause"
  limit: number; // default: 5
}
```

### Modules

_(New in version 7.13.0.)_

_Import name: `Sentry.Integrations.Modules`_

This integration fetches names of all currently installed Node modules and attaches the list to the event. Once fetched, Sentry will cache the list for later reuse.

## Modifying System Integrations

> To disable system integrations set `defaultIntegrations: false` when calling `init()`.

Reviewer: ```suggestion
To disable system integrations, set `defaultIntegrations: false` when calling `init()`.
```
